### PR TITLE
400 single showcase file

### PR DESCRIPTION
**Pull request by StaehliJ**
# Pull request

## Description

The goal of this PR is to simplify showcases by remove not necessarily files and improving codes.

## Changes made

- Self explanatory

## Checklist

- [x] Your branch has been rebased onto the `main` branch.
- [ ] APIs have been properly documented (if relevant).
- [ ] The documentation has been updated (if relevant).
- [ ] New unit tests have been written (if relevant).
- [ ] The demo has been updated (if relevant).
- [x] All pull request status checks pass.

### CODE REVIEW

File: pillarbox-demo/src/main/java/ch/srgssr/pillarbox/demo/ui/showcases/layouts/SimpleStory.kt [hunk 1]
```diff
@@ -51,29 +64,30 @@ fun SimpleStory() {
...
+        PlayerModule.provideDefaultPlayer(context).apply {
+            setMediaItem(demoItem.toMediaItem())
+            videoScalingMode = C.VIDEO_SCALING_MODE_SCALE_TO_FIT_WITH_CROPPING
+            repeatMode = Player.REPEAT_MODE_ONE
+        }
+    }
+    DisposableEffect(player) {
+        player.prepare()
+        onDispose {
+            player.release()
```
Comment (MGaetan89) [inline pillarbox-demo/src/main/java/ch/srgssr/pillarbox/demo/ui/showcases/layouts/SimpleStory.kt:78]: Why not do that in the `LifecycleStartEffect` block below?

File: pillarbox-demo/src/main/java/ch/srgssr/pillarbox/demo/ui/showcases/misc/MultiPlayerShowcase.kt [hunk 2]
```diff
@@ -30,13 +35,40 @@ import ch.srgssr.pillarbox.demo.ui.theme.paddings
...
+    LifecycleResumeEffect(Unit) {
+        playerOne.play()
+        playerTwo.play()
+        onPauseOrDispose {
+            playerOne.pause()
+            playerTwo.pause()
+        }
+    }
+    val leftPlayer = if (swapLeftRight) playerOne else playerTwo
+    val rightPlayer = if (swapLeftRight) playerTwo else playerOne
```
Comment (MGaetan89) [inline pillarbox-demo/src/main/java/ch/srgssr/pillarbox/demo/ui/showcases/misc/MultiPlayerShowcase.kt:71]: Just a detail, but shouldn't this be:
```suggestion
    val leftPlayer = if (swapLeftRight) playerTwo else playerOne
    val rightPlayer = if (swapLeftRight) playerOne else playerTwo
```

File: pillarbox-demo/src/main/java/ch/srgssr/pillarbox/demo/ui/showcases/misc/ResizablePlayerShowcase.kt [hunk 3]
```diff
@@ -53,11 +54,16 @@ fun AdaptivePlayerShowcase() {
...
         onDispose {
             player.release()
         }
     }
+    LifecycleStartEffect(player) {
+        player.play()
+        onStopOrDispose {
+            player.pause()
+        }
+    }
```
Comment (MGaetan89) [inline pillarbox-demo/src/main/java/ch/srgssr/pillarbox/demo/ui/showcases/misc/ResizablePlayerShowcase.kt:66]: This you're using start/stop (and not resume/pause), the `DisposableEffect` could be merged here too, no?
```suggestion
    LifecycleStartEffect(player) {
        player.prepare()
        player.play()
        onStopOrDispose {
            player.release()
        }
    }
```

File: pillarbox-demo/src/main/java/ch/srgssr/pillarbox/demo/ui/showcases/misc/UpdatableMediaItemShowcase.kt [hunk 4]
```diff
@@ -32,4 +33,10 @@ fun UpdatableMediaItemShowcase() {
             )
         }
     }
+    LifecycleResumeEffect(player) {
```
Comment (MGaetan89) [inline pillarbox-demo/src/main/java/ch/srgssr/pillarbox/demo/ui/showcases/misc/UpdatableMediaItemShowcase.kt:36]: Shouldn't we also `release` the player?

File: pillarbox-demo/src/main/java/ch/srgssr/pillarbox/demo/ui/player/SimplePlayerActivity.kt [hunk 5]
```diff
@@ -72,21 +72,17 @@ class SimplePlayerActivity : ComponentActivity(), ServiceConnection {
...
-            repeatOnLifecycle(Lifecycle.State.RESUMED) {
-                SRGAnalytics.trackPagView(DemoPageView("simple player", levels = listOf("app", "pillarbox")))
-            }
-        }
-        lifecycleScope.launchWhenCreated {
-            playerViewModel.pictureInPictureRatio.collectLatest {
-                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O && playerViewModel.pictureInPictureEnabled.value) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            lifecycleScope.launch {
+                playerViewModel.pictureInPictureRatio.flowWithLifecycle(lifecycle, Lifecycle.State.CREATED).collectLatest {
```
[the remark this answers is not in the document]
Comment (StaehliJ) [reply pillarbox-demo/src/main/java/ch/srgssr/pillarbox/demo/ui/player/SimplePlayerActivity.kt:77]: Yes we can change the parameters even if picture in picture is not enable.

File: pillarbox-demo/src/main/java/ch/srgssr/pillarbox/demo/ui/showcases/integrations/MediaControllerActivity.kt [hunk 6]
```diff
@@ -43,14 +43,9 @@ class MediaControllerActivity : ComponentActivity() {
...
-            repeatOnLifecycle(Lifecycle.State.RESUMED) {
-                SRGAnalytics.trackPagView(DemoPageView("media controller player", levels = listOf("app", "pillarbox")))
-            }
-        }
-        lifecycleScope.launchWhenCreated {
-            controllerViewModel.pictureInPictureRatio.collectLatest {
-                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O && controllerViewModel.pictureInPictureEnabled.value) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            lifecycleScope.launch {
+                controllerViewModel.pictureInPictureRatio.flowWithLifecycle(lifecycle, Lifecycle.State.CREATED).collectLatest {
```
[the remark this answers is not in the document]
Comment (StaehliJ) [reply pillarbox-demo/src/main/java/ch/srgssr/pillarbox/demo/ui/showcases/integrations/MediaControllerActivity.kt:48]: yes

File: pillarbox-demo/src/main/java/ch/srgssr/pillarbox/demo/ui/showcases/misc/ResizablePlayerShowcase.kt (hunk 3 above)
Comment (StaehliJ) [reply pillarbox-demo/src/main/java/ch/srgssr/pillarbox/demo/ui/showcases/misc/ResizablePlayerShowcase.kt:66]: As we don't want to release the player when activity is in stop, I would say no.

File: pillarbox-demo/src/main/java/ch/srgssr/pillarbox/demo/ui/showcases/misc/UpdatableMediaItemShowcase.kt (hunk 4 above)
Comment (StaehliJ) [reply pillarbox-demo/src/main/java/ch/srgssr/pillarbox/demo/ui/showcases/misc/UpdatableMediaItemShowcase.kt:36]: It is already released inside the associated ViewModel `UpdatableMediaItemViewModel`

**Review by MGaetan89 — APPROVED** [review]

**Pull request merged**